protoarray/src/main/java/tech/pegasys/artemis/protoarray/ProtoArrayForkChoiceStrategy.java

```diff
@@ -271,26 +272,20 @@ public boolean containsBlock(Bytes32 blockRoot) {
    * </ul>
    *
    * @param indices
-   * @param votes
+   * @param store
    * @param oldBalances
    * @param newBalances
    * @return
    */
   static List<Long> computeDeltas(
+      MutableStore store,
       Map<Bytes32, Integer> indices,
-      ElasticList<VoteTracker> votes,
       List<UnsignedLong> oldBalances,
       List<UnsignedLong> newBalances) {
     List<Long> deltas = new ArrayList<>(Collections.nCopies(indices.size(), 0L));
 
-    for (int validatorIndex = 0; validatorIndex < votes.size(); validatorIndex++) {
-      VoteTracker vote = votes.get(validatorIndex);
-
-      // There is no need to create a score change if the validator has never voted
-      // or both their votes are for the zero hash (alias to the genesis block).
-      if (vote.getCurrentRoot().equals(Bytes32.ZERO) && vote.getNextRoot().equals(Bytes32.ZERO)) {
-        continue;
-      }
+    for (int validatorIndex : store.getVotedValidatorIndices()) {
+      VoteTracker vote = store.getVote(validatorIndex);
```

Comment:
You'd be better off just iterating through the value set or entry set of the votes map since you need both the key and value here.

Also, `getVotedValidatorIndices` will return entries for every validator that we've ever retrieved - they may still have a current and next root of zero depending on whether anything was changed after it was retrieved.

Reply:
We only retrieve validators that have voted, because the ones that haven’t voted (i.e. the ones that have current and next root zero) doesn’t matter for protoarray fork choice. For the edge case where we retrieve a validator who had both roots zero, (this is the validators' first time attesting) since the validator never voted before, we have to change one of their roots to something non zero. Thus, all the votes `getVotedValidatorIndices` return must be non zero. @protolambda do you mind correcting me if I'm wrong?

Reply:
I'll still add the continue for votes that have both current and next root zero, in case someone else updates votes as discussed offline.

Reply:
@protolambda any feedback on this?

Reply:
looking into it now

Reply:
I don't think it actually matters too much, but an entry set would technically be better. Both for performance, as understanding the code (get all votes is easier to get right with consistency etc. than it is to get votes one by one). Small details though, good to see this all come together in teku :+1: